scripts/create-package.js

```diff
@@ -0,0 +1,183 @@
+const generate = argv.generate === undefined;
+
+// Convert any package names given in dash-case (e.g. my-new-package) to PascalCase (e.g. MyNewPackage)
+// for display purposes in certain template files (e.g. README.md)
+let pascalCasePackage = newPackageName.replace(/-[a-zA-Z]/g, function(m, t) {
+  return newPackageName[t + 1].toUpperCase();
+});
+pascalCasePackage = pascalCasePackage.substring(0, 1).toUpperCase() + pascalCasePackage.substring(1);
+
+// Current date used in CHANGELOG.md
+const today = new Date().toUTCString();
+
+// Paths
+const packagePath = `./packages/${newPackageName}`;
+const templateFolderPath = './scripts/templates/';
+const rushPath = __dirname + '/../rush.json';
+
+// rush.json contents
+let rushJson = require(rushPath);
+
+// Output Files
+const outputFiles = {
+  NpmIgnore: `${packagePath}/.npmignore`,
+  Npmrc: `${packagePath}/.npmrc`,
+  ChangelogJson: `${packagePath}/CHANGELOG.json`,
+  ChangelogMarkdown: `${packagePath}/CHANGELOG.md`,
+  License: `${packagePath}/LICENSE`,
+  Readme: `${packagePath}/README.md`,
+  IndexHtml: `${packagePath}/index.html`,
+  JestConfig: `${packagePath}/jest.config.js`,
+  JsConfig: `${packagePath}/jsconfig.json`,
+  PackageJson: `${packagePath}/package.json`,
+  TsConfig: `${packagePath}/tsconfig.json`,
+  TsLint: `${packagePath}/tslint.json`,
+  WebpackConfig: `${packagePath}/webpack.config.js`,
+  WebpackServeConfig: `${packagePath}/webpack.serve.config.js`,
+  VsCodeLaunch: `${packagePath}/.vscode/launch.json`,
+  VsCodeSettings: `${packagePath}/.vscode/settings.json`,
+  Tests: `${packagePath}/config/tests.js`,
+  ApiExtractorDisabled: `${packagePath}/config/api-extractor.json.disabled`,
+  TestsCommon: `${packagePath}/src/common/tests.js`,
+  AppDefinition: `${packagePath}/src/demo/AppDefinition.tsx`,
+  ColorStyles: `${packagePath}/src/demo/ColorStyles.scss`,
+  GettingStartedPageStyles: `${packagePath}/src/demo/GettingStartedPage.scss`,
+  GettingStartedPage: `${packagePath}/src/demo/GettingStartedPage.tsx`,
+  DemoStyles: `${packagePath}/src/demo/index.scss`,
+  Demo: `${packagePath}/src/demo/index.tsx`
+};
+
+// Strings
+const successCreatedPackage = `New package ${newPackageName} successfully created.`;
+const npmGenerateMessage = 'Running "npm generate" (to bypass this step, use --no-generate arg)';
+const rushPackagePresent = `Package @uifabric/${newPackageName} is already present in rush.json`;
+const errorUnableToCreatePackage = `Error creating package directory ${packagePath}`;
+const errorUnableToOpenTemplate = templateFile => `Unable to open mustache template ${templateFile} for component`;
+const errorUnableToWriteFile = step => `Unable to write ${step} file`;
+const errorUnableToUpdateRush = 'Unable to update rush.json';
+
+// Functions
+function handleError(error, errorPrependMessage) {
+  if (error) {
+    console.error(errorPrependMessage, error);
+    return false;
+  }
+
+  return true;
+}
+
+function performStep(stepIndex) {
```

Comment:
we have the `async` lib that allows you to write something really neat code with callbacks against an array: https://caolan.github.io/async/ it would simplify this a bunch...

Or maybe consider just doing readFileSync

Reply:
There were some issues with trying to use the `async` lib. This could be addressed in a future PR.